types/index.d.ts

```diff
@@ -0,0 +1,4 @@
+export function HomebaseProvider(props?: object): object;
+export function useTransact(): Array<Array<object>>;
+export function useEntity(lookup: object | string | number): Array<any>;
+export function useQuery(query: object, args?: object): Array<object>;
```

Comment:
We should type the JSONQuery and make the $find and $where attributes mandatory.
Then it can be `useQuery(query: JSONQuery | string, ...args: any)`
And it can return a Tuple where the first element is an Array of Entities

Reply:
Resolving though we didn't make the $find and $where attributes mandatory yet